docs/vm-configuration.md

```diff
@@ -266,3 +266,227 @@ the generation of multiple libvirt XML documents, the domain XML and associated
 documents for secrets, network filters and node devices. So there is not a strict
 1-1 mapping between the k8s API resource representation and the libvirt low
 level representation.
+
+Mapping rules for mapping libvirt XML to the KubeVirt API
+---------------------------------------------------------
+
+In order to support libvirt features, we tend to expose the libvirt API as much
+as possible in a 1:1 fashion. However, there are some differences which need to
+be taken into account:
+
+ 1. Our API is json/yaml and not xml.
+ 2. Host specifics are not exposed on the API.
+ 3. We want to make use of the golang xml/yaml/json mappers as much as possible.
+ 4. Ordering of devices can be important (e.g. for pci address assignment).
+ 5. KubeVirt also delivers specialized device types (mostly for network and
+    storage), which need to be fit into the existing structure.
+
+**Note**: Point **4** is still under debate.
+
+## General mapping rules
+
+### All devices need to live directly under *devices*
+
+In order to allow arbitrary ordering of devices, they all need directly live
+under `devices`.
+
+This is OK:
+
+```yaml
+devices:
+  lun:
+    my: data
+  video:
+    qxl:
+      my: data
+  lun:
+    my: data1
+  video:
+    qxl:
+      my: data1
+```
+
+This is **not** OK:
```

Comment:
Can you please elaborate, why this is not OK? I can imagine that switching the order of disks and videos would be non problematic.

Reply:
I think the difference here is not the ordering but the fact that the devices do not need to be grouped. Correct me if I'm wrong, though.

Reply:
@michalskrivanek wants to influence the order of devices, because of address defaulting. If they are grouped, then we have no absolute order of all PCI devices. @michalskrivanek something to add there?